s3_storage_provider.py

```diff
@@ -1,217 +1,316 @@
+        self.local_media_lifetime = None
+        self.remote_media_lifetime = None
+
+        # Parse boolean values from config
+        def parse_bool(value):
+            if value is None:
+                return None
+            if isinstance(value, bool):
+                return value
+            if isinstance(value, str):
+                return value.lower() in ("true", "yes", "1", "on")
+            return bool(value)
+
+        # Get settings directly from config
+        self.store_local = parse_bool(config.get("store_local"))
+        self.store_remote = parse_bool(config.get("store_remote"))
+        self.store_synchronous = parse_bool(config.get("store_synchronous"))
+
+        # Get media retention settings
+        media_retention = config.get("media_retention", {})
+        if isinstance(media_retention, dict):
+            self.local_media_lifetime = media_retention.get("local_media_lifetime")
+            self.remote_media_lifetime = media_retention.get("remote_media_lifetime")
+        else:
+            logger.warning("media_retention config is not a dictionary: %s", media_retention)
 
-        if "secret_access_key" in config:
-            self.api_kwargs["aws_secret_access_key"] = config["secret_access_key"]
+        logger.info(
+            "S3 Storage Provider initialized with store_local=%s, store_remote=%s, store_synchronous=%s",
+            self.store_local,
+            self.store_remote,
+            self.store_synchronous,
+        )
+        logger.info(
+            "Media retention settings: local=%s, remote=%s milliseconds",
+            self.local_media_lifetime,
+            self.remote_media_lifetime,
+        )
 
-        if "session_token" in config:
-            self.api_kwargs["aws_session_token"] = config["session_token"]
+        # S3 configuration
+        s3_config = config.get("config", {})
+        if not s3_config:
+            logger.warning("No config block found, falling back to root-level settings")
+            self.bucket = config.get("bucket")
+            self.api_kwargs = {
+                "region_name": config.get("region_name"),
+                "endpoint_url": config.get("endpoint_url"),
+                "aws_access_key_id": config.get("access_key_id"),
+                "aws_secret_access_key": config.get("secret_access_key"),
+            }
+            self.prefix = config.get("prefix", "")
+            self.extra_args = config.get("extra_args", {})
+        else:
+            self.bucket = s3_config.get("bucket")
+            self.api_kwargs = {
+                "region_name": s3_config.get("region_name"),
+                "endpoint_url": s3_config.get("endpoint_url"),
+                "aws_access_key_id": s3_config.get("access_key_id"),
+                "aws_secret_access_key": s3_config.get("secret_access_key"),
+            }
+            self.prefix = s3_config.get("prefix", "")
+            self.extra_args = s3_config.get("extra_args", {})
```

Comment:
`config` contains the contents of the `config` key. It won't contain a `config` key itself, so this `if` isn't necessary.